app/assets/javascripts/pretty-text/engines/discourse-markdown/upload-protocol.js

```diff
@@ -1,88 +1,151 @@
+import xss from "xss";
+
+const HTML_TYPES = ["html_block", "html_inline"];
+
 // add image to array if src has an upload
 function addImage(uploads, token) {
   if (token.attrs) {
     for (let i = 0; i < token.attrs.length; i++) {
       if (token.attrs[i][1].indexOf("upload://") === 0) {
-        uploads.push([token, i]);
+        uploads.push([token, i, token.attrs[i][1]]);
```

Comment:
I wonder if this would be clearer, if we used objects so that the keys are named.